### PR TITLE
Only guest customize if there is something to customize

### PR DESCRIPTION
**Pull request by jon-shanks**
We may not want to run guest customization on everything if there is no need to. Plus, not everything may be able to be customized i.e. coreos

### CODE REVIEW

**Comment by ajlanghorn** [comment]
@jon-shanks Thanks for the PR! There are three failures on this branch. Can you take a look at them, please, before we look to merge?

**Comment by joshmyers** [comment]
I can't push to that repo, I have fixed the failing tests. Please see:
https://github.com/joshmyers/vcloud-launcher/commit/9d13185b8de98085f72fbfdb0536505d603a0ef4
https://github.com/joshmyers/vcloud-launcher/commit/b9ddbb4afef473050279d3ef8fe3fed976e116a0

May need to cherry pick those off.

**Comment by allandegnan** [comment]
I cherry picked those, thanks @joshmyers

File: lib/vcloud/launcher/vm_orchestrator.rb [hunk 1]
```diff
@@ -23,7 +23,7 @@ def customize(vm_config)
 
         preamble = vm_config[:bootstrap] ? generate_preamble(vm_config) : ''
 
-        @vm.configure_guest_customization_section(preamble)
+        @vm.configure_guest_customization_section(preamble) if (!preamble.nil? && !preamble.empty?)
```
Comment (ajlanghorn) [inline lib/vcloud/launcher/vm_orchestrator.rb]: What happens when the preamble value is literally set to `nil`? Can a test be added for that?
It's an edge case, but it's possible that it could occur.

Comment (mattbostock) [inline lib/vcloud/launcher/vm_orchestrator.rb]: @joshmyers Is the check for `nil` necessary? Are there any instances under which `preamble` would be `nil`?

If so, the logic here could be simplified to:

```ruby
@vm.configure_guest_customization_section(preamble) unless preamble.nil? || preamble.empty?
```

Comment (joshmyers) [inline lib/vcloud/launcher/vm_orchestrator.rb]: @mattbostock see below while running tests:

```ruby
pry(#<Vcloud::Launcher::VmOrchestrator>)>
.
From: /Users/joshmyers/projects/vcloud-launcher/lib/vcloud/launcher/vm_orchestrator.rb @ line 25 Vcloud::Launcher::VmOrchestrator#customize:

    10: def customize(vm_config)
    11:   @vm.update_name(@vm.vapp_name)
    12:   @vm.configure_network_interfaces vm_config[:network_connections]
    13:   @vm.update_storage_profile(vm_config[:storage_profile]) if vm_config[:storage_profile]
    14:   if vm_config[:hardware_config]
    15:     @vm.update_cpu_count(vm_config[:hardware_config][:cpu])
    16:     @vm.update_memory_size_in_mb(vm_config[:hardware_config][:memory])
    17:   end
    18:   @vm.add_extra_disks(vm_config[:extra_disks])
    19:   @vm.update_metadata(vm_config[:metadata])
    20:   if vm_config.key?(:independent_disks)
    21:     Vcloud::Launcher::IndependentDiskOrchestrator.new(@vm).attach(vm_config.fetch(:independent_disks))
    22:   end
    23:
    24:   preamble = vm_config[:bootstrap] ? generate_preamble(vm_config) : ''
 => 25:   require 'pry';binding.pry
    26:   @vm.configure_guest_customization_section(preamble) if (!preamble.nil? && !preamble.empty?)
    27: end

[1] pry(#<Vcloud::Launcher::VmOrchestrator>)> preamble
=> nil
[2] pry(#<Vcloud::Launcher::VmOrchestrator>)> vm_config
=> {:hardware_config=>{:memory=>4096, :cpu=>2},
 :network_connections=>[{:name=>"network1", :ip_address=>"198.12.1.21"}],
 :bootstrap=>{:script_path=>"/tmp/bootstrap.erb", :vars=>{:message=>"hello world"}}}
[3] pry(#<Vcloud::Launcher::VmOrchestrator>)> generate_preamble(vm_config)
=> nil
[4] pry(#<Vcloud::Launcher::VmOrchestrator>)> preamble.empty?
NoMethodError: undefined method `empty?' for nil:NilClass
from (pry):10:in `customize'
[5] pry(#<Vcloud::Launcher::VmOrchestrator>)> preamble.nil?
=> true
```

Comment (joshmyers) [inline lib/vcloud/launcher/vm_orchestrator.rb]: :+1: for the more readable logic though!

Comment (joshmyers) [inline lib/vcloud/launcher/vm_orchestrator.rb]: @allandegnan would you be able to change to use @mattbostock logic?

**Comment by jon-shanks** [comment]
sorry was away on holiday, thanks for the changes guys. modified the logic and pushed up to be inline the tests pass with the amendments locally.

**Comment by mattbostock** [comment]
Thanks @jon-shanks and @joshmyers!
